node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala

```diff
@@ -59,39 +61,55 @@ case class NeutrinoNode(
     * @return
     */
   override def sync(): Future[Unit] = {
-    val blockchainsF =
-      BlockHeaderDAO()(executionContext, chainConfig).getBlockchains()
     for {
       chainApi <- chainApiFromDb()
       header <- chainApi.getBestBlockHeader()
-      filterHeaderCount <- chainApi.getFilterHeaderCount()
+      bestFilterHeaderOpt <- chainApi.getBestFilterHeader()
       filterCount <- chainApi.getFilterCount()
-      blockchains <- blockchainsF
-    } yield {
+      blockchains <-
+        BlockHeaderDAO()(executionContext, chainConfig).getBlockchains()
       // Get all of our cached headers in case of a reorg
-      val cachedHeaders = blockchains.flatMap(_.headers).map(_.hashBE.flip)
-      peerMsgSender.sendGetHeadersMessage(cachedHeaders)
-
-      // If we have started syncing filters headers
-      if (filterHeaderCount != 0) {
-        peerMsgSender.sendNextGetCompactFilterHeadersCommand(
-          chainApi = chainApi,
-          filterHeaderBatchSize = chainConfig.filterHeaderBatchSize,
-          prevStopHash = header.hashBE)
```

Comment:
this is the bug, it doesn't make sicne to use our best block header's block hash. We need to use our best filter header's block hash to make sure filter headers are contiguous